src/SAML2/Provider/HostedIdentityProviderProcessor.php

```diff
@@ -446,14 +464,14 @@ public function continueSingleLogoutService()
             $this->stateHandler->apply(SamlStateHandler::TRANSITION_SLS_START_PROPAGATE);
 
             // Dispatch logout to service providers
-            $sp = $this->serviceProviderRepository->getServiceProvider($state->popServiceProviderIds());
+            $sp = $this->serviceProviderRepository->getServiceProvider($state->getRequest()->getIssuer());
```

Comment:
The idea here is to send LogoutRequest to all service providers attached to the account. This will propagate logout to all service provider. 

I think the logout propagation could be an option ?